File: Easy/Rust/184.rs
// Daily Coding Problem: 184
// Difficulty: Easy
// Asked by: Amazon
// Authored by: TenthEdict

fn gcd(a: i32, b: i32) -> i32 {
    let r = a % b;
    if r == 0 {
        b
    } else {
        gcd(b, r)
    }
}

fn gcd_of_array(arr: &[i32]) -> i32 {
    let mut result = arr[0];
    for i in 1..arr.len() {
        result = gcd(result, arr[i])
    }
    result
}

fn main() {
    let numbers = [42, 56, 14];
    println!("{}", gcd_of_array(&numbers)); // returns 14 !
}

File: Python/Easy/184.py
# Daily Coding Problem: 184
# Difficulty: Easy
# Asked by: Amazon
# Authored by: TenthEdict

def gcd(a, b):
    r = a % b
    if r == 0:
        return b
    else:
        return gcd(b, r)

def gcd_of_array(arr):
    result = arr[0]
    for i in range(len(arr)):
        result = gcd(result, arr[i])
    return result

def main():
    numbers = [42,56,14]
    print(gcd_of_array(numbers))

main()